doc/man/man4/SIGTRAN.4.man
'\" rtp
.\" vim: ft=nroff sw=4 noet nocin nosi com=b\:.\\\" fo+=tcqlorn tw=77
.\" =========================================================================
.\"
.\" @(#) doc/man/man4/SIGTRAN.4.man
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2008-2011  Monavacon Limited <http://www.monavacon.com/>
.\" Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
.\" Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to copy, distribute and/or modify this manual under
.\" the terms of the GNU Free Documentation License, Version 1.3 or any later
.\" version published by the Free Software Foundation; with no Invariant
.\" Sections, no Front-Cover Texts, and no Back-Cover Texts.  A copy of the
.\" license is included in the section entitled "GNU Free Documentation
.\" License".
.\"
.\" Permission to use, copy and distribute this manual without modification,
.\" for any purpose and without fee or royalty is hereby granted, provided
.\" that both the above copyright notice and this permission notice appears
.\" in all copies and that the name of OpenSS7 Corporation not be used in
.\" advertising or publicity pertaining to distribution of this documentation
.\" or its contents without specific, written prior permission.  OpenSS7
.\" Corporation makes no representation about the suitability of this manual
.\" for any purpose.  It is provided "as is" without express or implied
.\" warranty.
.\"
.\" Permission is granted to process this file through groff and print the
.\" results, provided the printed document carries a copying permission
.\" notice identical to this one except for the removal of this paragraph
.\" (this paragraph not being relevant to the printed manual).
.\"
.\" OPENSS7 CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS MANUAL
.\" INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
.\" PARTICULAR PURPOSE, NON-INFRINGEMENT, OR TITLE; THAT THE CONTENTS OF THE
.\" DOCUMENT ARE SUITABLE FOR ANY PURPOSE, OR THAT THE IMPLEMENTATION OF SUCH
.\" CONTENTS WILL NOT INFRINGE ON ANY THIRD PARTY PATENTS, COPYRIGHTS,
.\" TRADEMARKS OR OTHER RIGHTS.  IN NO EVENT SHALL OPENSS7 CORPORATION BE
.\" LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES OR ANY
.\" DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
.\" IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
.\" OUT OF OR IN CONNECTION WITH ANY USE OF THIS DOCUMENT OR THE PERFORMANCE
.\" OR IMPLEMENTATION OF THE CONTENTS THEREOF.
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this manual
.\" page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from the
.\" use of the information contained herein.  The author(s) may not have
.\" taken the same level of care in the production of this manual, which is
.\" licensed free of charge, as they might when working professionally.  The
.\" author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by the
.\" source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
.\" behalf of the U.S. Government ("Government"), the following provisions
.\" apply to you.  If the Software is supplied by the Department of Defense
.\" ("DoD"), it is classified as "Commercial Computer Software" under
.\" paragraph 252.227-7014 of the DoD Supplement to the Federal Acquisition
.\" Regulations ("DFARS") (or any successor regulations) and the Government
.\" is acquiring only the license rights granted herein (the license rights
.\" customarily provided to non-Government users).  If the Software is
.\" supplied to any unit or agency of the Government other than DoD, it is
.\" classified as "Restricted Computer Software" and the Government's rights
.\" in the Software are defined in paragraph 52.227-19 of the Federal
.\" Acquisition Regulations ("FAR") (or any successor regulations) or, in the
.\" cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the
.\" FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database openss7.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
search-truncate 40
search-ignore CGIQOSTU
.R2
.so openss7.macros
.\"
.\"
.TH SIGTRAN 4 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ Devices"
.\"
.\"
.SH NAME
.B SIGTRAN
\- OpenSS7 Signalling Transport (SIGTRAN) Devices
.\"
.\"
.SH SYNOPSIS
.PP
.\"
.\"
.HP
.BI "int " sigtran_stream " = open(" sigtran_device ", " flags );
.\"
.\"
.SH DESCRIPTION
.PP
.\"
.\"
.SS "SS7 Devices"
.PP
.B SIGTRAN
.[
rfc2719
.]
provides the following SS7 pseudo-device drivers:
.\"
.\"
.TP
.BR bua (4)
.I SS7 BICC User Adaptation Layer (BUA)
protocol layer transporting and distributing the services provided by
.BR bicc (4)
and providing an IO Control interface,
.BR lmi_ioctl (4),
for BUA-specific management,
.BR bua_ioctl (4).
.\"
.\"
.TP
.BR isua (4)
.I SS7 ISUP User Adaptation Layer (ISUA)
.[
isua-00
.]
protocol layer transporting and distributing the services provided by
.BR isup (4)
and providing an IO Control interface,
.BR lmi_ioctl (4),
for ISUA-specific management,
.BR isua_ioctl (4).
.\"
.\"
.TP
.BR tua (4)
.I SS7 TCAP User Adaptation Layer (TUA)
.[
tua-01
.]
protocol layer transporting and distributing the services provided by
.BR tcap (4)
and providing an IO Control interface,
.BR lmi_ioctl (4),
for TUA-specific management,
.BR tua_ioctl (4).
.\"
.\"
.TP
.BR sua (4)
.I SS7 SCCP User Adaptation Layer (SUA)
.[
rfc3868
.]
protocol layer transporting and distributing the services provided by
.BR sccp (4)
and providing an IO Control interface,
.BR lmi_ioctl (4),
for SUA-specific management,
.BR sua_ioctl (4).
.\"
.\"
.TP
.BR m3ua (4)
.I SS7 MTP3 User Adaptation Layer (M3UA)
.[
rfc3332
.]
.[
ts 102 142
.]
.[
rfc4666
.]
protocol layer transporting and distributing the services provided by
.BR mtp (4)
and providing an IO Control interface,
.BR lmi_ioctl (4),
for M3UA-specific management,
.BR m3ua_ioctl (4).
.\"
.\"
.TP
.BR m2ua (4)
.I SS7 MTP2 User Adaptation Layer (M2UA)
.[
rfc3331
.]
.[
ts 102 141
.]
protocol layer transporting and distributing the services provided by
.BR sl (4)
and providing an IO Control interface,
.BR lmi_ioctl (4),
for M2UA-specific management,
.BR m2ua_ioctl (4).
.\"
.\"
.SS "SS7 Modules"
.PP
.B SIGTRAN
provides the following SS7 modules:
.\"
.\"
.TP
.BR m2pa (4)
.I SS7 MTP2 User Peer-to-Peer Adaptation Layer (M2PA)
.[
M2PArfc
.]
.\"
.\"
.SS "ISDN Devices"
.PP
.B SIGTRAN
provides the following ISDN pseudo-device drivers:
.\"
.\"
.TP
.BR iua (4)
.I ISDN Q.921 User Adaptation Layer (IUA)
.[
IUArfc
.]
protocol layer transporting and distributing the services provided by
.BR dl (4),
and providing an IO Control Interface,
.BR lmi_ioctl (4),
for IUA-specific management,
.BR iua_ioctl (4).
.\"
.\"
.TP
.BR dua (4)
.I DPNSS/DASS 2 Extentions to the IUA Protocol (DUA)
.[
DUArfc
.]
protocol layer transporting and distributing the services provided by
.BR dua (4),
and providing an IO Control Itnerface,
.BR lmi_ioctl (4),
for DUA-specific management,
.BR dua_ioctl (4).
.\"
.\"
.TP
.BR v5ua (4)
.I V5 User Adaptation Layer (V5UA)
.[
V5UA-RFC
.]
protocol layer transporting and distributing the services provided by
.BR v5 (4),
and providing an IO Control Interface,
.BR lmi_ioctl (4),
for V5UA-specific management,
.BR v5ua_ioctl (4).
.\"
.\"
.TP
.BR gr303ua (4)
.I GR-303 User Adaptation Layer (GR303UA)
.[
gr303ua00
.]
protocol layer transporting and distributing the services provided by
.BR gr303 (4),
and providing an IO Control Interface,
.BR lmi_ioctl (4),
for GR303UA-specific management,
.BR gr303ua_ioctl (4).
.\"
.\"
.\".SH NOTICES
.PP
.\"
.\"
.\".SH BUGS
.PP
.\"
.\"
.\".SH FILES
.PP
.\"
.\"
.SH DEVICES
.PP
.IR /dev/bua ,
.IR /dev/isua ,
.IR /dev/tua ,
.IR /dev/tua-tpi-cl ,
.IR /dev/tua-tpi-co ,
.IR /dev/sua ,
.IR /dev/sua-npi ,
.IR /dev/sua-tpi-cl ,
.IR /dev/sua-tpi-co ,
.IR /dev/m3ua ,
.IR /dev/m3ua-npi ,
.IR /dev/m3ua-tpi-cl ,
.IR /dev/m3ua-tpi-co ,
.IR /dev/m2ua ,
.IR /dev/m2ua-cdi ,
.IR /dev/m2ua-dlpi ,
.IR /dev/iua ,
.IR /dev/v5ua ,
.IR /dev/gr303ua .
.\"
.\"
.SH "SEE ALSO"
.PP
.BR ss7confd (8),
.BR isdnconfd (8),
.BR SS7 (4),
.BR ISDN (4),
.BR MG (4).
.\"
.\"
.SH VERSIONS
.PP
This manpage was updated @PACKAGE_DATE@ for
.B @PACKAGE@
release
.BR "@VERSION@" .
.PP
Most of the devices and modules listed here do not appear until
.BR strss7-0.8.4 .
.PP
.BR m2pa (4)
first appeared in
.BR strss7-0.8.2 .
.\"
.[
$LIST$
.]
.\"
.TI


NEWS
OpenSS7 -- history of user visible changes.  2015-01-15
$Id: NEWS,v 1.1.2.25 2011-05-10 13:45:30 brian Exp $
Copyright (c) 2008-2015  Monavacon Limited. <http://www.monavacon.com/>
Copyright (c) 2001-2008  OpenSS7 Corporation. <http://www.openss7.com/>
Copyright (c) 1997-2001  Brian Bidulock <bidulock@openss7.org>
See the end for copying conditions (for this file).

The sections that follow provide information on OpenSS7 releases of the
'OpenSS7' package.

Major changes for release openss7-1.1.7.20150115
------------------------------------------------

This is the seventh release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Source code release on GitHub.

Major changes for release openss7-1.1.7.20131209
------------------------------------------------

This is the seventh release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Build corrections for RHEL 6.4.

   - Scripts for country boundary, shoreline and rivers database
     preparation.

   - Database for ANSI T1.101 point code assignments.

   - Suppress extransous /proc/1/comm error messages when testing for
     systemd.

   - Handle PUIAS to Springdale rename for PUIAS distribution.

Major changes for release openss7-1.1.7.20131123
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Correction to ca-cert handling in RHEL 6 install scriptlet.

   - RHEL 6 build changes.

   - Set permissions correctly when applying kernel module patch.

   - Do not duplicate certs when including ca-cert.

   - Avoid conflicting shelll variable in RPM install scriplet.

   - Remove /proc/1/comm checks for systemd in module loader.

Major changes for release openss7-1.1.7.20130209
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Corrections to RPM spec file.

Major changes for release openss7-1.1.7.20130129
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Compensate for RHEL6 lack of c_rehash.

   - Do not create sysfs nodes: they do not work properly.

   - Use no foreground flag in strace.service file.

   - More error traces for strerr logger.

   - Remove static device names (the do not work either).

   - Changes to support weak kernel modules.

   - Clean up weak module support.

Major changes for release openss7-1.1.7.20130125
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Better Archlinux install script support.

Major changes for release openss7-1.1.6.20130125
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Remove use of kill_litter_super().

Major changes for release openss7-1.1.3.20130125
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Updates and corrections to weak kernel module builds.

   - Repo support and corrections for Archlinux packages.

Major changes for release openss7-1.1.3.20130123
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Additional support for Archlinux.

   - Add udev rules for strace and strerr.

   - Add system unit files for strace and strerr.

   - Add repo support for Archlinux packages.

   - Build Archlinux packages better.

Major changes for release openss7-1.1.3.20130121
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Build support for SLES 11.

   - Handle certificates better.

   - Get configuration files in the right places for SLES 11.

   - Support in RPM spec file for certificates on SLES 11.

   - Create a new OpenSS7 certificate.

   - Repository support for SLES 11.

   - RPM spec file reworked for new RPM release.

   - Better certificate handling for RHEL.

   - Additional build support for RHEL 6 with certificates.


src/util/init/sigtran.sh
#!/bin/sh
#
# @(#) src/util/init/sigtran.sh
# Copyright (c) 2008-2015  Monavacon Limited <http://www.monavacon.com>
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
# All Rights Reserved.
#
# Distributed by OpenSS7 Corporation.  See the bottom of this script for copying
# permissions.
#
# These are arguments to update-rc.d ala chkconfig and lsb.  They are recognized
# by openss7 install_initd and remove_initd scripts.  Each line specifies
# arguments to add and remove links after the the name argument:
#
# sigtran:	start and stop sigtran modules
# update-rc.d:	start 33 S . stop 33 0 6 .
# config:	/etc/default/sigtran
# probe:	false
# hide:		false
# license:	GPL
# description:	This STREAMS init script is part of Linux Fast-STREAMS.  \
#		It is responsible for ensuring that the necessary STREAMS \
#		SIGTRAN character devices are present in the /dev directory and \
#		that the STREAMS SIGTRAN kernel modules are configured and loaded.
#
# LSB init script conventions
#
### BEGIN INIT INFO
# Provides: sigtran
# Required-Start: streams strcompat $network
# Required-Stop: streams strcompat $network
# Default-Start: 3 4 5
# Default-Stop: 0 1 2 6
# X-UnitedLinux-Default-Enabled: yes
# Short-Description: start and stop STREAMS Binary Compatibility Modules
# License: GPL
# Description:	This STREAMS SIGTRAN init script is part of Linux Fast-STREAMS.
#	It is reponsible for ensuring that the necessary STREAMS SIGTRAN character
#	devices are present in the /dev directory and that the STREAMS SIGTRAN
#	kernel modules are configured and loaded.
### END INIT INFO

PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
name='sigtran'
config="/etc/default/$name"
desc="the STREAMS SIGTRAN modules"
mknod="${name}_mknod"

[ -e /proc/modules ] || exit 0

if test -z "$SIGTRAN_MKNOD" ; then
    for SIGTRAN_MKNOD in /sbin/${mknod} /usr/sbin/${mknod} /bin/${mknod} /usr/bin/${mknod} ; do
	if [ -x $SIGTRAN_MKNOD ] ; then
	    break
	else
	    SIGTRAN_MKNOD=
	fi
    done
fi

# Specify defaults

[ -n "$SIGTRAN_PRELOAD"       ] || SIGTRAN_PRELOAD=""
[ -n "$SIGTRAN_DRIVERS"       ] || SIGTRAN_DRIVERS="streams-ua_as streams-m2ua_as streams-m3ua_as streams-sua_as streams-tua_as streams-isua_as"
[ -n "$SIGTRAN_MODULES"       ] || SIGTRAN_MODULES="streams-m2pa_sl streams-sdl_sctp streams-sdt_sctp streams-sdt_tpi streams-sl_tpi streams-sl_sctp streams-m2tp"
[ -n "$SIGTRAN_MAKEDEVICES"   ] || SIGTRAN_MAKEDEVICES="yes"
[ -n "$SIGTRAN_REMOVEDEVICES" ] || SIGTRAN_REMOVEDEVICES="yes"

# Source config file
for file in $config ; do
    [ -f $file ] && . $file
done

[ -z "$SIGTRAN_MKNOD" ] && SIGTRAN_MAKEDEVICES="no"
[ -z "$SIGTRAN_MKNOD" ] && SIGTRAN_REMOVEDEVICES="no"

RETVAL=0

umask 077

case :$VERBOSE in
    :no|:NO|:false|:FALSE|:0|:)
	redir='>/dev/null 2>&1'
	;;
    *)
	redir=
	;;
esac

build_options() {
    # Build up the options string
    :
}

start() {
    echo -n "Loading STREAMS kernel modules: "
    RETVAL=0
    modules=
    for module in $SIGTRAN_PRELOAD ; do
	modules="${modules:+$modules }$module"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if ! eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -q -- $module $redir"
	    [ $? -eq 0 ] || echo -n "(failed)"
	fi
    done
    echo "."

    echo -n "Starting $desc: $name "
    build_options
    if [ $? -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
	RETVAL=1
    fi

    if eval "grep '^[[:space:]]*${name}[/.]' /etc/sysctl.conf $redir" ; then
	echo -n "Reconfiguring kernel parameters: "
	eval "sysctl -p /etc/sysctl.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	fi
    fi

    if [ -f /etc/${name}.conf ] ; then
	echo -n "Configuring STREAMS parameters: "
	eval "sysctl -p /etc/${name}.conf $redir"
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi

    if [ -n "$SIGTRAN_MKNOD" -a :"$SIGTRAN_MAKEDEVICES" = ":yes" ] ; then
	echo -n "Making STREAMS SIGTRAN devices: "
	$SIGTRAN_MKNOD
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    return $RETVAL
}

stop() {
    echo "Stopping $desc: $name "
    RETVAL=0
    if [ -n "$SIGTRAN_MKNOD" -a ":$SIGTRAN_REMOVEDEVICES" = ":yes" ] ; then
	echo -n "Removing STREAMS devices: "
	$SIGTRAN_MKNOD --remove
	if [ $? -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	    RETVAL=1
	fi
    fi
    echo -n "Unloading STREAMS kernel modules: "
    modules=
    for module in $SIGTRAN_PRELOAD $SIGTRAN_DRIVERS $SIGTRAN_MODULES ; do
	modules="$module${modules:+ $modules}"
    done
    for module in $modules ; do
	modrex=`echo $module | sed -e 's,[-_],[-_],g'`
	if eval "grep '^$modrex\>' /proc/modules $redir" ; then
	    echo -n "$module "
	    eval "modprobe -r -q -- $module $redir"
	    if [ $? -ne 0 ] ; then
		echo -n "(failed) "
		RETVAL=1
	    fi
	fi
    done
    if [ $RETVAL -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
    fi
    return $RETVAL
}

restart() {
    stop
    start
    return $?
}

show() {
    echo "$name.sh: show: not yet implemented." >&2
    return 1
}

usage() {
    echo "Usage: /etc/init.d/$name.sh (start|stop|restart|force-reload|show)" >&2
    return 1
}

case "$1" in
    (start|stop|restart|show)
	$1 || RETVAL=$?
	;;
    (force-reload)
	restart || RETVAL=$?
	;;
    (*)
	usage || RETVAL=$?
	;;
esac

[ "${0##*/}" = "$name.sh" ] && exit $RETVAL

# =============================================================================
# 
# @(#) src/util/init/sigtran.sh
#
# -----------------------------------------------------------------------------
#
# Copyright (c) 2008-2015  Monavacon Limited <http://www.monavacon.com/>
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
#
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or modify it under
# the terms of the GNU Affero General Public License as published by the Free
# Software Foundation; version 3 of the License.
#
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more
# details.
#
# You should have received a copy of the GNU Affero General Public License along
# with this program.  If not, see <http://www.gnu.org/licenses/>, or write to
# the Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
#
# -----------------------------------------------------------------------------
#
# U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
# behalf of the U.S. Government ("Government"), the following provisions apply
# to you.  If the Software is supplied by the Department of Defense ("DoD"), it
# is classified as "Commercial Computer Software" under paragraph 252.227-7014
# of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
# successor regulations) and the Government is acquiring only the license rights
# granted herein (the license rights customarily provided to non-Government
# users).  If the Software is supplied to any unit or agency of the Government
# other than DoD, it is classified as "Restricted Computer Software" and the
# Government's rights in the Software are defined in paragraph 52.227-19 of the
# Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
# the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
# (or any successor regulations).
#
# -----------------------------------------------------------------------------
#
# Commercial licensing and support of this software is available from OpenSS7
# Corporation at a fee.  See http://www.openss7.com/
#
# =============================================================================
# vim: ft=sh sw=4 tw=80
